### ISSUE TITLE
ProfileHelper used with UserProfile raises a ClassNotFoundException

### ISSUE THREAD

**Issue by rafamarquezv** [pac4j/pac4j#500]
- Artifacts: pac4j-jwt & pac4j-http (1.8.6)
- Class: org.pac4j.core.profile.ProfileHelper
- Method: determineProfileClassByName

**Description**:
When trying to determinate profile class for UserProfile the returned "completeName" string references to a wrong package. It returns "org.pac4j.oauth.profile.user.UserProfile" when it should return "org.pac4j.core.profile.UserProfile", so a java.lang.ClassNotFoundException is raised later in "buildUserProfileByClassCompleteName" method on line 116.

**Used code to reproduce**:
```

        UserProfile up = new UserProfile();
        up.setId("id");
        up.setRemembered(false);
        up.addAttribute("attribute_1", "value_1");
        up.addAttribute("attribute_2", "value_2");
        up.addAttribute("attribute_3", "value_3");
        up.addPermission("Permission_1");
        up.addPermission("Permission_2");
        up.addRole("Role_1");
        up.addRole("Role_2");

        JwtGenerator<UserProfile> jwtGeneratorUp =
            new JwtGenerator<>(signingSecret, encryptionSecret);

        jwtGeneratorUp.setEncryptionMethod(EncryptionMethod.A128CBC_HS256);

        String tokenUp = jwtGeneratorUp.generate(up);
        System.out.println("GENERATED TOKEN: " + tokenUp);

        UserProfile up2 = jwtAuthenticator.validateToken(tokenUp);
        System.out.println("RETRIEVED INFO: " + up2.toString());
```
_Note: "signingSecret" and "encryptionSecret" must be declared properly._

**Comment by leleuj** [comment]
I see the problem. This is a limitation of the "typed identifier" which only refers to the class name and not to the full class name with package. So the full class name is computed at best.
In pac4j v1.9, the "typed identifier" is now the full class name with package which makes profile rebuilding always working: https://github.com/pac4j/pac4j/blob/master/pac4j-core/src/main/java/org/pac4j/core/profile/ProfileHelper.java#L73

So I consider this problem solved in pac4j 1.9. As a workaround for pac4j v1.8, I would recommend using the `HttpProfile` instead of the `UserProfile`.

**Comment by rafamarquezv** [comment]
Great! Thank you.
